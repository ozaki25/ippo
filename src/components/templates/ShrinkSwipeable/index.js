import React from 'react';
import SwipeableViews from 'react-swipeable-views';
import propTypes from 'prop-types';

const styles = {
  root: {
    padding: '0 15px',
  },
  slideContainer: {
    padding: '0 5px',
    width: 'calc(100% - 10px)',
  },
};

const ShrinkSwipeable = ({ children }) => (
  <SwipeableViews
    style={styles.root}
    slideStyle={styles.slideContainer}
    hysteresis={0.4}
    enableMouseEvents
  >
    {children}
  </SwipeableViews>
);

ShrinkSwipeable.displayName = 'ShrinkSwipeable';

ShrinkSwipeable.propTypes = {
  children: propTypes.node.isRequired,
};

export default ShrinkSwipeable;
