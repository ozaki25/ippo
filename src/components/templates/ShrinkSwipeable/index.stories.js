import React from 'react';
import ShrinkSwipeable from '.';

export default {
  title: 'templates/ShrinkSwipeable',
};

const style = {
  padding: 15,
  backgroundColor: 'lightblue',
};

const items = [
  <h1 style={style} key="1">
    テスト1
  </h1>,
  <h1 style={style} key="2">
    テスト2
  </h1>,
  <h1 style={style} key="3">
    テスト3
  </h1>,
  <h1 style={style} key="4">
    テスト4
  </h1>,
  <h1 style={style} key="5">
    テスト5
  </h1>,
];

export const 通常パターン = () => <ShrinkSwipeable children={items} />;
