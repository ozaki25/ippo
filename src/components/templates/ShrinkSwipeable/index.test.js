import React from 'react';
import { snapshot } from 'src/test/helpers';
import ShrinkSwipeable from '.';

const props = {
  children: [
    <h1 key="1">テスト1</h1>,
    <h1 key="2">テスト2</h1>,
    <h1 key="3">テスト3</h1>,
    <h1 key="4">テスト4</h1>,
    <h1 key="5">テスト5</h1>,
  ],
};

snapshot('ShrinkSwipeable/nomal', <ShrinkSwipeable {...props} />);
