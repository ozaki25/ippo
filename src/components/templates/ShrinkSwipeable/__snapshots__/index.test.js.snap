// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: ShrinkSwipeable/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <ShrinkSwipeable>
    <ReactSwipableView
      animateHeight={false}
      animateTransitions={true}
      axis="x"
      disableLazyLoading={false}
      disabled={false}
      enableMouseEvents={true}
      hysteresis={0.4}
      ignoreNativeScroll={false}
      index={0}
      resistance={false}
      slideStyle={
        Object {
          "padding": "0 5px",
          "width": "calc(100% - 10px)",
        }
      }
      springConfig={
        Object {
          "delay": "0s",
          "duration": "0.35s",
          "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
        }
      }
      style={
        Object {
          "padding": "0 15px",
        }
      }
      threshold={5}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onScroll={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflowX": "hidden",
            "padding": "0 15px",
          }
        }
      >
        <div
          className="react-swipeable-view-container"
          style={
            Object {
              "WebkitFlexDirection": "row",
              "WebkitTransition": "all 0s ease 0s",
              "direction": "ltr",
              "display": "flex",
              "flexDirection": "row",
              "height": null,
              "transition": "all 0s ease 0s",
              "willChange": "transform",
            }
          }
        >
          <div
            aria-hidden={false}
            data-swipeable="true"
            key=".$1"
            style={
              Object {
                "WebkitFlexShrink": 0,
                "flexShrink": 0,
                "overflow": "auto",
                "padding": "0 5px",
                "width": "calc(100% - 10px)",
              }
            }
          >
            <h1
              key="1"
            >
              テスト1
            </h1>
          </div>
        </div>
      </div>
    </ReactSwipableView>
  </ShrinkSwipeable>
</MockRouter>
`;
